network_simulator/link.py

```diff
@@ -59,28 +101,24 @@ class Link(object):
     specified, but not the Devices themselves (since those are references).
     """
 
-    def __init__(self, end_1_addr=None, end_2_addr=None, end_1_device=None,
-                 end_2_device=None, link_buffer=LinkBuffer(),
-                 static_delay_sec=None, capacity_bps=None):
+    def __init__(self, end_1_device=None, end_2_device=None,
+            link_buffer=LinkBuffer(), static_delay_sec=None,
+            capacity_bps=None, busy = False):
         """
-        :ivar string end_1_addr: address of Device on one end (e.g. "H1").
```

Comment:
Don't get rid of end_1_addr and end_2_addr. I need them for reading in from JSON, since we cannot read in Devices from JSON.